server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WithStorageDirective.java

```diff
@@ -61,12 +64,14 @@ public WithStorageDirective(UsersRepository usersRepository) {
 
     @Override
     public void init() throws MessagingException {
-        targetFolderName = getInitParameter(TARGET_FOLDER_NAME);
-        validateMailetConfiguration();
+        targetFolderName = AttributeValue.of(validateMailetConfiguration(TARGET_FOLDER_NAME));
+        validateMailetConfiguration(getInitParameter(TARGET_FOLDER_NAME));
```

Comment:
You do a `getInitParamater` as well in `validateMailetConfiguration`. No need of doing it twice right?